Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -749,7 +749,8 @@ void HTMLInputElement::attributeChanged(const QualifiedName& name, const AtomStr
 
     switch (name.nodeName()) {
     case AttributeNames::typeAttr:
-        updateType();
+        if (oldValue != newValue)
```

Comment:
I'm surprised we don't have a global check early returning for this case.

Reply:
We definitely don't currently and I see this function getting called with the same type attribute value in the context of Speedometer. We may be able to have a more global check though. I can investigate and follow-up if it works out.